sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -352,8 +352,7 @@ object functions {
    * @group agg_funcs
    * @since 1.3.0
    */
-  def count(columnName: String): TypedColumn[Any, Long] =
-    count(Column(columnName)).as(ExpressionEncoder[Long]())
+  def count(columnName: String): Column = count(Column(columnName))
```

Comment:
This seems to me it is wrongly being a `TypedColumn`. `Count` is a `DeclarativeAggregate`.

Reply:
This is a breaking change, right?

At lease https://github.com/apache/spark/pull/27499/files#diff-2c67e6ae3d5115b5521681f6ef871b1dR43 is broken.

Reply:
It seems a right change but let's revert this line considering it's code freeze period ..

Reply:
Ok. :)